kafkalog/kafka.go

```diff
@@ -197,38 +224,18 @@ func Init(hosts []string, duration, queueLen int) error {
 		go func() {
 			ticker := time.NewTicker(time.Duration(duration) * time.Second)
 			for _ = range ticker.C {
+				kp.initAsyncProducer(hosts)
 				glog.V(3).Infof("producer %v: length of chan: %v, %v producers %v",
 					kp, len(kp.logChan), number, kp.producer)
-
-				kp.Lock()
-				if kp.producer == nil {
-					kp.Unlock()
-
-					producer, err := NewAsyncProducer(hosts)
-					if err != nil {
-						glog.Errorf("[producer]: fail to connect to kafka: %v", err)
-						continue
-					} else {
-						glog.Infof("Connected to kafka")
-					}
-
-					kp.Lock()
-					if kp.producer == nil {
-						kp.producer = producer
-					} else {
-						// Can not happen
-						glog.V(3).Infof("kafka producer %v alreay connected", kp)
-						go func() {
-							producer.Close()
-						}()
-					}
-				}
-				kp.Unlock()
 			}
 		}()
+
+		if err := kp.initAsyncProducer(hosts); err != nil {
+			retErr = err
+		}
 	}
 
-	return nil
+	return retErr
```

Comment:
hyperd will notice the fail and log warning message.
https://github.com/getdvm/hyper-private/pull/180